### PR TITLE
Search with overlapped regex.

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is fixed.

- Update serch algorithm to cover overlapping regex case
- Add more separators pattern (PHP, tyization case)

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTest
- [x] Benchmark

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -22,7 +22,8 @@ jobs:
     - name: Checkout CredData
       uses: actions/checkout@v3
       with:
-        repository: Samsung/CredData
+        repository: babenek/CredData
+        ref: auxiliary
```
Comment (babenek) [inline .github/workflows/benchmark.yml:26]:
```suggestion
        repository: Samsung/CredData
```

File: .github/workflows/benchmark.yml [hunk 2]
```diff
@@ -62,7 +63,8 @@ jobs:
...
```
Comment (babenek) [inline .github/workflows/benchmark.yml:67]:
```suggestion
        repository: Samsung/CredData
```

**Review by babenek — COMMENTED** [review]
TODO: rollback after https://github.com/Samsung/CredData/pull/65

**Review by Yullia — APPROVED** [review]
Approved

**Review by iuriimet — APPROVED** [review]

**Pull request merged**